### DISCUSSION TITLE
About Kyuubi and HiveServer2

### DISCUSSION THREAD

**Discussion by kissycn** [apache/kyuubi#5133] [Q&A] [answered]
It may be that I understand the deviation. I know from some documents and materials that Kyuubi can replace HiveServer2
We are now using Hive On Spark SQL gateway using HiveServer2, I wonder if HiveServer2 is not needed after switching to Kyuubi, my HQL client is directly connected to Kyuubi
But it is not tested in practice to connect to Kyuubi via HQL
So what I want to ask is:
1. Can Kyuubi completely replace HiveServer2 to parse the HQL of Hive On Spark?
2. If I want to migrate to Kyuubi, do I have to change my HiveSQL to SparkSQL, and HiveSerer2 should also run synchronously until all migrations are completed.

**Comment by yaooqinn** [comment]
What is HQL short for？

**Reply by kissycn** [reply]
@yaooqinn  HiveSQL

**Reply by yaooqinn** [reply, marked as answer]
The answer for q1 is yes. And for q2, the incompatibility between Hive QL and Spark SQL, both syntax and some implicit behavior, does exist but few. It shouldn't be a problem.

**Comment by kissycn** [comment]
@yaooqinn  Thank you for your answer, and kyuubi is really cool
